src/main/java/io/aiven/connect/jdbc/dialect/PostgreSqlDatabaseDialect.java

```diff
@@ -234,9 +254,41 @@ protected String getSqlType(final SinkRecordField field) {
                 return "TEXT";
             case BYTES:
                 return "BYTEA";
+            case ARRAY:
+                return getSqlTypeFromSchema(schema.valueSchema()) + "[]";
             default:
-                return super.getSqlType(field);
+                return null;
+        }
+    }
+
+    @Override
+    protected boolean maybeBindPrimitive(
+            final PreparedStatement statement,
+            final int index,
+            final Schema schema,
+            final Object value
+    ) throws SQLException {
+        if (schema.type() == Schema.Type.ARRAY) {
+            final Schema.Type valueType = schema.valueSchema().type();
+            final Class<?> componentType = SUPPORTED_ARRAY_VALUE_TYPES_TO_JAVA.get(valueType);
+            if (componentType != null) {
+                List<?> valueCollection = (List<?>) value;
+                // Postgres does not have an 8-bit integer, using short as the best alternative.
+                if (valueType == Schema.Type.INT8) {
+                    valueCollection = valueCollection.stream()
+                            .map(o -> ((Byte) o).shortValue()).collect(Collectors.toList());
+                }
+                final Object newValue = Array.newInstance(componentType, valueCollection.size());
+                for (int i = 0; i < valueCollection.size(); i++) {
+                    Array.set(newValue, i, valueCollection.get(i));
+                }
+                statement.setObject(index, newValue, Types.ARRAY);
+                return true;
+            } else {
+                throw new DataException(String.format("Unsupported schema type %s for ARRAY values", valueType));
+            }
         }
```

Comment:
Add `else` here? Or invent the whole thing - the idea is that if it's not an array, let super handle it. Otherwise handle it in method.